crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -68,7 +68,7 @@ type CheckpointExecutionBuffer = FuturesOrdered<JoinHandle<VerifiedCheckpoint>>;
 /// The interval to log checkpoint progress, in # of checkpoints processed.
 const CHECKPOINT_PROGRESS_LOG_COUNT_INTERVAL: u64 = 5000;
 
-const SCHEDULING_EVENT_FUTURE_TIMEOUT_MS: u64 = 1200;
+const SCHEDULING_EVENT_FUTURE_TIMEOUT_MS: u64 = 2000;
```

Comment:
Based on the frequency of the errors, I wonder if the threshold needs to be more conservative even for `warn`, e.g. 10s ~ 30s. Usually network timeouts are > 5s for things that need to be eagerly retried. And on validators, do they process the complete sequence of checkpoints via StateSync as well?

Reply:
@mwtian yes, for both fullnodes and validators, state sync is processing the entire checkpoint sequence. If the checkpoint contents already exist locally as a result of consensus, it is a no-op, but state sync still increments the watermark and notifies checkpoint executor.

The assumption here is that a checkpoint should have been generated once per second as consensus guarantees that (correct me if I'm wrong), so anything greater than 1 second should be sufficient. The version of the code that is causing errors uses a wait time of exactly one second, giving rise to a race condition.

I am happy to move to 5 seconds still if we want, but let me run this version run in replay tests to ensure that we indeed no longer see the log line since I already merged

Reply:
Makes sense. Indeed checkpoints should be generated with 1s interval in the ideal case. But with down consensus leaders or internet disruptions (affecting consensus or state sync), the intervals can get longer. If needed, we can also deploy this to private testnet and turn on logging at warn level to see how frequent the logs are. Ultimately you can decide on what warning interval would be the most useful for you.